File: scripts/algorithms/C/Check If It Is a Good Array/Check If It Is a Good Array.cpp
// Runtime: 100 ms (Top 27.69%) | Memory: 29.4 MB (Top 37.30%)
class Solution {
public:
    bool isGoodArray(vector<int>& nums) {
        int gcd=0;
        for(int i=0; i<nums.size(); i++){
            gcd=__gcd(gcd,nums[i]);
        }return gcd==1;
    }
};

File: scripts/algorithms/C/Check If It Is a Good Array/Check If It Is a Good Array.js
// Runtime: 150 ms (Top 20.00%) | Memory: 48.4 MB (Top 40.00%)

var isGoodArray = function(nums) {
    let gcd = nums[0]

    for(let n of nums){while(n){[gcd, n] = [n, gcd % n]}}

    return (gcd === 1)
};